interpreter/value_fix128.go

```diff
@@ -0,0 +1,652 @@
+	panic(errors.NewUnreachableError())
+}
+
+func (Fix128Value) SetMember(_ ValueTransferContext, _ LocationRange, _ string, _ Value) bool {
+	// Numbers have no settable members (fields / functions)
+	panic(errors.NewUnreachableError())
+}
+
+func (v Fix128Value) ToBigEndianBytes() []byte {
+	fix128 := fix.Fix128(v)
+	return fixedpoint.Fix128ToBigEndianBytes(fix128)
+}
+
+func (v Fix128Value) ConformsToStaticType(
+	_ ValueStaticTypeConformanceContext,
+	_ LocationRange,
+	_ TypeConformanceResults,
+) bool {
+	return true
+}
+
+func (Fix128Value) IsStorable() bool {
+	return true
+}
+
+func (v Fix128Value) Storable(_ atree.SlabStorage, _ atree.Address, _ uint64) (atree.Storable, error) {
+	return v, nil
+}
+
+func (Fix128Value) NeedsStoreTo(_ atree.Address) bool {
+	return false
+}
+
+func (Fix128Value) IsResourceKinded(_ ValueStaticTypeContext) bool {
+	return false
+}
+
+func (v Fix128Value) Transfer(
+	context ValueTransferContext,
+	_ LocationRange,
+	_ atree.Address,
+	remove bool,
+	storable atree.Storable,
+	_ map[atree.ValueID]struct{},
+	_ bool,
+) Value {
+	if remove {
+		RemoveReferencedSlab(context, storable)
+	}
+	return v
+}
+
+func (v Fix128Value) Clone(_ ValueCloneContext) Value {
+	return v
+}
+
+func (Fix128Value) DeepRemove(_ ValueRemoveContext, _ bool) {
+	// NO-OP
+}
+
+func (v Fix128Value) ByteSize() uint32 {
+	fix128 := fix.Fix128(v)
+
+	// tag number (2 bytes) + array head (1 byte) + high-bits (CBOR uint) + low-bits (CBOR uint)
+	return values.CBORTagSize +
+		1 +
+		values.GetUintCBORSize(uint64(fix128.Hi)) +
+		values.GetUintCBORSize(uint64(fix128.Lo))
+}
+
+func (v Fix128Value) StoredValue(_ atree.SlabStorage) (atree.Value, error) {
+	return v, nil
+}
+
+func (Fix128Value) ChildStorables() []atree.Storable {
+	return nil
+}
+
+func (v Fix128Value) IntegerPart() NumberValue {
+	// TODO: Maybe compute this without the use of `big.Int`.
+	fix128BigInt := v.ToBigInt()
+
+	integerPart := new(big.Int).Div(fix128BigInt, sema.Fix128FactorIntBig)
+
+	// The max length of the integer part is 128-bits.
+	// Therefore, return an `Int128`.
+	return NewUnmeteredInt128ValueFromBigInt(integerPart)
+}
+
+func (Fix128Value) Scale() int {
+	return sema.Fix128Scale
+}
+
+func (v Fix128Value) ToBigInt() *big.Int {
+	return fixedpoint.Fix128ToBigInt(fix.Fix128(v))
+}
+
+func handleFixedpointError(err error, _ LocationRange) {
+	if err == nil {
+		return
+	}
+
+	// TODO: handle/wrap external error with a cadence overflow/underflow error.
+	panic(err)
+}
+
+func performSaturationArithmatic(
+	operation func(value fix.Fix128) (fix.Fix128, error),
+	otherValue Fix128Value,
+) fix.Fix128 {
+	result, err := operation(fix.Fix128(otherValue))
+
+	// Should not panic on overflow/underflow
+
+	// TODO: Switch on error type, rather than the value.
+	// 	Need changes to the fixedpoint library.
+	switch err {
+	case fix.ErrOverflow, fix.ErrNegOverflow:
+		return fix.Fix128Max
+	case fix.ErrUnderflow:
+		return fix.Fix128Min
```

Comment:
I use "underflow" to mean "the magnitude is too small to be represented", and "negative overflow" to mean "the result is negative and has a magnitude too large to be represented". (And, of course, "overflow" means "the result is positive, and the magnitude is too large to be represented".)

```suggestion
	case fix.ErrOverflow:
		return fix.Fix128Max
	case fix.ErrNegOverflow:
		return fix.Fix128Min
	case fix.ErrUnderflow:
		return fix.Fix128Zero // Or however you represent 0.0
```